backend/app/controllers/books_controller.rb

```diff
@@ -5,16 +5,18 @@ class BooksController < ApplicationController
   before_action :authorize_request
 
   def index
-    render(json: CollectionPresenter.new(books), status: :ok)
+    render(json: CollectionPresenter.new(books, BookSerializer), status: :ok)
   end
 
   def show
-    render(json: { data: book }, status: :ok)
+    render(json: { data: book ? BookSerializer.new(book) : nil }, status: :ok)
```

Comment:
nit, if u can put the nil handling in the presenter the better so u won't have to put a ternary here. This'll make it consistent to your other endpoints in terms of cleanliness.